Core/Src/steeringio.c

```diff
@@ -93,16 +97,58 @@ void steeringio_update(steeringio_t* wheel, uint8_t wheel_data[], uint8_t len)
 
 	// TODO: Revisit how new wheel's data is formatted
 	/* Copy message data to wheelio buffer */
-	// memcpy(&io, wheel_data, len);
+	osMutexAcquire(wheel->button_mutex, osWaitForever);
+	// Data is formatted with each bit within the first byte representing a button and the first two bits of the second byte representing the paddle shifters
+
+	/* Update raw buttons */
+	for (uint8_t i = 0; i < MAX_STEERING_BUTTONS - 2; i++) {
+		wheel->raw_buttons[i] = (wheel_data[0] >> i) & 0x01;
+	}
+
+	/* Update raw paddle shifters */
+	wheel->raw_buttons[STEERING_PADDLE_LEFT] = (wheel_data[1] >> 0) & 0x01;
+	wheel->raw_buttons[STEERING_PADDLE_RIGHT] = (wheel_data[1] >> 1) & 0x01;
 
 	/* If the value was set high and is not being debounced, trigger timer for debounce */
-	osMutexAcquire(wheel->button_mutex, osWaitForever);
 	for (uint8_t i = 0; i < MAX_STEERING_BUTTONS; i++) {
-		if (wheel->debounced_buttons[i] && wheel->raw_buttons[i]) {
+		if (!wheel->debounced_buttons[i] && wheel->raw_buttons[i]) {
 			wheel->debounced_buttons[i] = PRESSED;
+			printf("%d index pressed \r\n",i);
+			switch (i) {
+				case STEERING_PADDLE_LEFT:
+					// doesnt effect cerb for now
+					break;
+				case STEERING_PADDLE_RIGHT:
+					// doesnt effect cerb for now
+					break;
+				case NERO_BUTTON_UP:
+					serial_print("Up button pressed \r\n");
+					decrement_nero_index();
+					break;
+				case NERO_BUTTON_DOWN:
+					serial_print("Down button pressed \r\n");
+					increment_nero_index();
+					break;
+				case NERO_BUTTON_LEFT:
+					// doesnt effect cerb for now
+					break;
+				case NERO_BUTTON_RIGHT:
+					// doesnt effect cerb for now
+					break;
+				case NERO_BUTTON_SELECT:
+					printf("Select button pressed \r\n");
+					select_nero_index();
+					break;
+				case NERO_HOME:
+					printf("Home button pressed \r\n");
+					set_home_mode();
+					break;
+				default:
+					break;
+			}
 		} else if (wheel->raw_buttons[i] && !osTimerIsRunning(wheel->debounce_timers[i])) {
-			osTimerStart(wheel->debounce_timers[i], STEERING_WHEEL_DEBOUNCE);
-			ringbuffer_enqueue(wheel->debounce_buffer, &i);
+			// osTimerStart(wheel->debounce_timers[i], STEERING_WHEEL_DEBOUNCE);
```

Comment:
I know that the debounce timer didn't work but we should revisit in the future when we have time. Here is a video on _electrical_ debouncing https://www.youtube.com/watch?v=e1-kc04jSE4

Ngl I didn't know that web dev people used the same term for a different problem